tests/test_auth.py

```diff
@@ -24,18 +25,42 @@ def test_auth_factory_bad_config():
         auth = AuthFactory.create(config)
 
 
-def test_auth_factory_provided_jwt():
+def test_auth_factory_provided_jwt_default_credentials():
     config = {'AUTH_TOKEN': 'abcd-1234'}
     auth = AuthFactory.create(config)
     assert isinstance(auth, ProvidedJwtTokenAuth)
 
     session = Session()
+    session.batch_mode = False
     auth.configure_session(session)
 
     assert session.headers.get('Authorization')
     assert session.headers['Authorization'] == 'Bearer abcd-1234'
 
 
+def test_auth_factory_provided_jwt_batch_credentials():
+    config = {'AUTH_TOKEN': 'abcd-1234', 'BATCH_MODE': {'AUTH_TOKEN': 'batch-abcd-1234'}}
+    auth = AuthFactory.create(config)
+    session = Session()
+    session.batch_mode = True
+    auth.configure_session(session)
+
+    assert session.headers.get('Authorization')
+    assert session.headers['Authorization'] == 'Bearer batch-abcd-1234'
+
+
+def test_auth_factory_provided_jwt_no_batch_credentials():
+    config = {'AUTH_TOKEN': 'abcd-1234'}
+    auth = AuthFactory.create(config)
+    session = Session()
+    session.batch_mode = True
+
+    try:
```

Comment:
When testing whether an exception is raised, use the `pytest.raises` context manager:

```python
with pytest.raises(ConfigError):
    auth.configure_session(session)
```

This will make sure that an exception is raised, and that it is the correct type.